### PR TITLE
Add bullet about citation to affiliated package guidelines.

### PR DESCRIPTION
**Pull request by kelle**
Add words about including citation instructions and citing other relevant works. Includes a link to the AstroBetter post about citing software.

### CODE REVIEW

File: affiliated/index.html [hunk 1]
```diff
@@ -255,6 +255,10 @@ <h1 id="becoming-an-affiliated-package">Becoming an Affiliated Package<a class="
 		consistent with basic principles of open source. (E.g.,
 		<a href="https://choosealicense.com/">an OSI-approved license</a>)</li>
 
+		<li>Include instructions to users on how to cite your package and include citations to
```
Comment (pllim) [inline affiliated/index.html:258]: Same comment as https://github.com/astropy/astropy-project/pull/229#discussion_r774681134

File: affiliated/index.html [hunk 2]
```diff
@@ -255,6 +255,10 @@ <h1 id="becoming-an-affiliated-package">Becoming an Affiliated Package<a class="
 		consistent with basic principles of open source. (E.g.,
 		<a href="https://choosealicense.com/">an OSI-approved license</a>)</li>
 
+		<li>Include instructions to users on how to cite your package. (This is commonly done
+			with a CITATION.md file.) Also include citations to other relevant papers and software following
```
Comment (dhomeier) [inline affiliated/index.html:259]:
```suggestion
			with a CITATION file.) Also include citations to other relevant papers and software following
```
The core package itself and all coordinated packages I have looked up are using either plain name or `CITATION.rst`;-) if we want to endorse .md (not actually sure what this implies for format or representation), we should at least, or most, put `CITATION or CITATION.md` at this point.

**Comment by kelle** [comment]
Are there any more comments on the words? Are we just waiting on the coordinated packages to add CITATION files?

File: affiliated/index.html [hunk 3]
```diff
@@ -255,6 +255,15 @@ <h1 id="becoming-an-affiliated-package">Becoming an Affiliated Package<a class="
 		consistent with basic principles of open source. (E.g.,
 		<a href="https://choosealicense.com/">an OSI-approved license</a>)</li>
 
+		<li>Include instructions to users on how to cite your package. This is commonly done
+			with a CITATION file. This file could include acknowledgement text, a Zenodo record
+			(highly recommended), and/or journal article(s). Where possible, full BibTeX entries of these citations
```
Comment (pllim) [inline affiliated/index.html:260]: Nitpick: Highly recommended option should be listed first.
```suggestion
			with a CITATION file. This file could include a Zenodo record (highly recommended),
			acknowledgement text, and/or journal article(s). Where possible, full BibTeX entries of these citations
```

**Review by pllim — APPROVED** [review]

**Review by dhomeier — APPROVED** [review]
Thanks all, looks good to me now. And congratulations on the Fellowship Kelle!
For my part I'd be OK with not waiting any longer for `astropy-healpix` and `reproject` to come up with their files; if they are seen to be aware of the issue and working on it, that should be good enough.

**Review by eteq — APPROVED** [review]
LGTM too, merging!

**Pull request merged**